### PR TITLE
Switch to a new addon folder structure, and update binary filenames

### PR DESCRIPTION
**Pull request by Ivorforce**
This PR does multiple things:
- Use build/addons/gdexample/ instead of bin to place the files in.
- Make proper use of prefix and suffix for the library binaries (this changes some of their names). 
    - Windows builds erroneously had the `lib` prefix, even though they weren't configured to.
        - This caused us trouble with release builds as we were shuffling around files.
    - macOS / ios builds were lacking the .dylib suffix.
        - I think is what caused https://github.com/godotengine/godot/issues/96403
- Adjust the corresponding .gdextension paths.
- The `.gdextension` file included in the root `addons/` folder points both debug and release to the release binary. This is because the github runner only builds release binaries. If the binary is not used for both, the published `.gdextension` cannot be used in the editor.

**Why use `build/addons/gdexample`?**
1) The godot asset library places assets under the root folder. I've heard people say it's good practice to place your assets in `/addons/xxx`, so we'd better start adopting some standard. I'm open to other suggestions. By using this subfolder structure already in build/, we can simply pack it in the git runner and it will pack it into a zip that's ready as-is for the asset store.
2) Given 'bin' contains a few files that are not built, but actually placed there by hand, I was inclined to rename it.

This PR supercedes #44 and #43.

### CODE REVIEW

**Comment by Ivorforce** [comment]
If we agree this is the file structure that we recommend, I will make a PR to adjust the godot-cpp intro tutorial in the docs accordingly.

File: .gitignore [hunk 1]
```diff
@@ -49,3 +50,8 @@ compile_commands.json
 .vscode/*
 !.vscode/extensions.json
 .DS_Store
+
+# Build binaries
+/build/addons/numdot/**/*
+!/build/addons/numdot/**/
+!/build/addons/numdot/**/*.plist
```
Comment (paddy-exe) [inline .gitignore:57]: I assume this change is not meant to be included in this PR?

**Review by paddy-exe — CHANGES_REQUESTED** [review]
I think this should be changed as it doesn't seem to belong to this PR. The rest of the changes are more substantial so I would wait for the next GDExtension meeting to talk about this PR. From a grand overview I think this looks good.

File: .gitignore (hunk 1 above)
Comment (Ivorforce) [reply .gitignore:57]: Oh, right, that should be `/build/addons/gdexample/`. Fixing

**Comment by dsnopek** [comment]
We discussed this at the GDExtension meeting, and aren't convinced that it makes sense to restructure to put the output into the addons directory in the extension source code. Not all GDExtensions use this layout, even for reusable GDExtensions, but there's also folks who use GDExtension for their gameplay logic where "addons/" may not make sense. But, also, you wouldn't point the Godot asset library directly at the extension source code - you'd either have a separate repo or make a ZIP file that contains the built GDExtension, so if you wanted to put it under "addons/", you could copy it there at that point.

**Comment by Ivorforce** [comment]
Ok, that's fine. I will make new PRs for the less contentious parts of this PR. Thanks for discussing!

**Comment by unvermuthet** [comment]
This is the structure I came up with for my project:

- `/demo` folder and scons `env.Copy` are gone
- Whole `/addons` folder is included in the build artifact
- Allows for easy Asset Store upload without having to manually copy around files
- `project.godot` allows you to have a project included with the repo for development
- Not included in build artifact
- Users that install extension can have demo scenes directly in _their_ project

```
addons/
└─ godot-ndi/
   ├─ bin/
   │  ├─ linux/
   │  │  ├─ .gitkeep
   │  │  └─ libgodot-ndi.linux.template_debug.x86_64.so
   │  ├─ macos/
   │  │  ├─ .gitkeep
   │  │  └─ godot-ndi.framework/
   │  │     └─ Resources/
   │  │        └─ Info.plist
   │  └─ windows/
   │     ├─ .gitkeep
   │     └─ godot-ndi.windows.template_debug.x86_64.dll
   ├─ icons/
   │  └─ # Class icons
   ├─ demo/
   │  └─ # Demo scenes
   ├─ project.godot # To have a bootable demo project in the repo, but ignored in build artifacts
   ├─ godot-ndi.gdextension
   └─ godot-ndi.gdextension.uid
```

This is great for extensions but not useful for gameplay logic applications.

**Pull request closed without merging**